<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] py-amqplib doesn't like Rabbit user refusals
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=%5Brabbitmq-discuss%5D%20py-amqplib%20doesn%27t%20like%20Rabbit%20user%20refusals&In-Reply-To=">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="002877.html">
   <LINK REL="Next"  HREF="002844.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] py-amqplib doesn't like Rabbit user refusals</H1>
    <B>Jason J. W. Williams</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=%5Brabbitmq-discuss%5D%20py-amqplib%20doesn%27t%20like%20Rabbit%20user%20refusals&In-Reply-To="
       TITLE="[rabbitmq-discuss] py-amqplib doesn't like Rabbit user refusals">jasonjwwilliams at gmail.com
       </A><BR>
    <I>Mon Feb  2 22:32:58 GMT 2009</I>
    <P><UL>
        <LI>Previous message: <A HREF="002877.html">[rabbitmq-discuss] Clustering Q
</A></li>
        <LI>Next message: <A HREF="002844.html">[rabbitmq-discuss] py-amqplib doesn't like Rabbit user refusals
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#2842">[ date ]</a>
              <a href="thread.html#2842">[ thread ]</a>
              <a href="subject.html#2842">[ subject ]</a>
              <a href="author.html#2842">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Hi,

It appears that py-amqplib really doesn't like it when you try to log
on to a vhost with a user that lacks authorization. Instead of
specific exception, a struct.error exception is raised with this
traceback:

2009-02-02 22:13:17+0000 [-] Traceback (most recent call last):
2009-02-02 22:13:17+0000 [-]   File &quot;zzzz.py&quot;, line 377, in yyyy
2009-02-02 22:13:17+0000 [-]     self.mq_connect()
2009-02-02 22:13:17+0000 [-]   File &quot;zzzz.py&quot;, line 145, in mq_connect
2009-02-02 22:13:17+0000 [-]     connect_timeout=5)
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/client_0_8.py&quot;, line 298,
in __init__
2009-02-02 22:13:17+0000 [-]     (10, 30), # tune
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/client_0_8.py&quot;, line 183,
in wait
2009-02-02 22:13:17+0000 [-]     frame_type, payload = self._next_frame()
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/client_0_8.py&quot;, line 123,
in _next_frame
2009-02-02 22:13:17+0000 [-]     return
self.connection._wait_channel(self.channel_id)
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/client_0_8.py&quot;, line 414,
in _wait_channel
2009-02-02 22:13:17+0000 [-]     frame_type, frame_channel, payload =
self._wait_frame()
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/client_0_8.py&quot;, line 395,
in _wait_frame
2009-02-02 22:13:17+0000 [-]     frame_type = self.input.read_octet()
2009-02-02 22:13:17+0000 [-]   File
&quot;/usr/lib64/python2.5/site-packages/amqplib/util_0_8.py&quot;, line 96, in
read_octet
2009-02-02 22:13:17+0000 [-]     return unpack('B', self.input.read(1))[0]
2009-02-02 22:13:17+0000 [-]   File &quot;/usr/lib64/python2.5/struct.py&quot;,
line 87, in unpack
2009-02-02 22:13:17+0000 [-]     return o.unpack(s)
2009-02-02 22:13:17+0000 [-] error: unpack requires a string argument
of length 1


The RabbitMQ side reports a rather sane error:

{channel0_error,starting,
                {amqp,access_refused,&quot;login refused for user 'argus'&quot;,
                      'connection.start_ok'}}


Is py-amqplib just not set up to cleanly handle user refusals, or is
this an unknown bug?

-J


</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="002877.html">[rabbitmq-discuss] Clustering Q
</A></li>
	<LI>Next message: <A HREF="002844.html">[rabbitmq-discuss] py-amqplib doesn't like Rabbit user refusals
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#2842">[ date ]</a>
              <a href="thread.html#2842">[ thread ]</a>
              <a href="subject.html#2842">[ subject ]</a>
              <a href="author.html#2842">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
